src/Lean/Data/Lsp/Basic.lean

```diff
@@ -206,7 +206,7 @@ instance : FromJson DocumentChange where
 [reference](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceEdit) -/
 structure WorkspaceEdit where
   /-- Changes to existing resources. -/
-  changes : RBMap DocumentUri TextEditBatch compare := ∅
+  changes? : Option (RBMap DocumentUri TextEditBatch compare) := none
```

Comment:
This was a bug in the implementation of the LSP spec: the `changes` and other fields are optional, and they should not be passed along unless they are being used. This was causing issues where `documentChanges` would be preferred over `changes` even though it is empty.